src/SnapKeyring.test.ts

```diff
@@ -353,6 +366,94 @@ describe('SnapKeyring', () => {
       spy.mockRestore();
     });
 
+    describe('async request redirect url', () => {
+      it('throws an error if async request redirect url is not an allowed origin', async () => {
+        const redirect = {
```

Comment:
Looks like those 2 tests are identical, so either we could rewrite this like:

```typescript
it.each([
  ['https://allowed.com'],
  [],
})(...)
```

Or maybe have a helper function that we call in both those functions like:
```typescript
const isNotAllowed = (allowedOrigins) => { ... };

it('throws an error if async request redirect url is not an allowed origin', async () => {
  isNotAllowed(['https://allowed.com']);
})

it('throws an error if no allowed origins', async () => {
  isNotAllowed([]);
})
```

I don't have a strong opinion, but having clear text messages is a bit better.

Reply:
I like this idea, and it simplifies the tests. I am in favor of keeping the clear text messages for the "it" blocks so I went with your suggestion to use a helper function in this commit [f1bc7ff](https://github.com/MetaMask/eth-snap-keyring/pull/228/commits/f1bc7ff5e2a4b2487e5b5d9c6ad1c0c2ef09d40c) lmk what you think please!